src/v0/destinations/x_audience/transform.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const {
+  removeUndefinedAndNullAndEmptyValues,
+  InstrumentationError,
+  ConfigurationError,
+} = require('@rudderstack/integrations-lib');
+const { handleRtTfSingleEventError } = require('../../util');
+const { batchEvents, buildResponseWithJSON, getUserDetails } = require('./utils');
+/**
+ * This function returns audience object in the form of destination API
+ * @param {*} message
+ * @param {*} destination
+ * @param {*} metadata
+ */
+const processRecordEvent = (message, config) => {
+  const { fields, action, type } = message;
+  if (type !== 'record') {
+    throw new InstrumentationError(`[X AUDIENCE]: ${type} is not supported`);
+  }
+  const { accountId, audienceId } = { config };
+  if (accountId) {
+    throw new ConfigurationError('[X AUDIENCE]: Account Id not found');
+  }
+  if (audienceId) {
+    throw new ConfigurationError('[X AUDIENCE]: Audience Id not found');
+  }
+  const { effective_at, expires_at } = fields;
```

Comment:
We should add validation for `effective_at` and `expires_at` otherwise anything apart from date may be mapped.

Reply:
For Type check we are clearly mentioning in docs.
Didn't want to do anything as this is through VDM so just wanted to send the as it is